src/aux-vm-browser/vm/AuxVMImpl.ts

```diff
@@ -237,6 +238,32 @@ export class AuxVMImpl implements AuxVM {
      */
     async sendEvents(events: BotAction[]): Promise<void> {
         if (!this._proxy) return null;
+        if (events && events.length) {
+            const transferables = [];
+            for (let event of events) {
+                if ((event as any).result) {
+                    event = <AsyncResultAction>event;
+
+                    if (event.result instanceof OffscreenCanvas) {
+                        console.log(
+                            `[AuxVMImpl] marked OffscreenCanvas as transferable in AsyncResultAction`,
+                            event
+                        );
+                        transferables.push(event.result);
+                    }
+                }
+            }
+
+            if (transferables.length > 0) {
+                console.log(
+                    `[AuxVMImpl] sendEvents marking transferables from events`,
+                    events,
+                    transferables
+                );
+                transfer(events, transferables);
```

Comment:
@Blitzy `transfer()` is a function that marks a value as containing a list of transferable objects. It doesn't actually transfer the values itself. You need to use the return value of `transfer()` and pass that to `this._proxy.sendEvents()` instead of `events`.